core/editable.js

```diff
@@ -1231,30 +1233,34 @@
 				this.editor.setData( this.editor.getData(), 0, 1 );
 
 				this.clearListeners();
-				this.restoreAttrs();
-
-				// Cleanup our custom classes.
-				var classes;
-				if ( ( classes = this.removeCustomData( 'classes' ) ) ) {
-					while ( classes.length )
-						this.removeClass( classes.pop() );
-				}
-
-				// Remove contents stylesheet from document if it's the last usage.
-				if ( !this.is( 'textarea' ) ) {
-					var doc = this.getDocument(),
-						head = doc.getHead();
-					if ( head.getCustomData( 'stylesheet' ) ) {
-						var refs = doc.getCustomData( 'stylesheet_ref' );
-						if ( !( --refs ) ) {
-							doc.removeCustomData( 'stylesheet_ref' );
-							var sheet = head.removeCustomData( 'stylesheet' );
-							sheet.remove();
-						} else {
-							doc.setCustomData( 'stylesheet_ref', refs );
+
+				catchEdgePermissionDenied( CKEDITOR.tools.bind( function() {
```

Comment:
As you move here line with `removeClass` method, it should be probably more readable to move `try/catch` clause into `this.detach` method but extract `try` code as a part of private `_.detach`. Something like:
 
```javascript
detach: function() {
	this.status = 'detached';

	// Update the editor cached data with current data.
	this.editor.setData( this.editor.getData(), 0, 1 );

	this.clearListeners();

	// Edge randomly throws permission denied when trying to access native elements of detached editor (#3115).
	try {
		this._.detach();
	} catch ( error ) {
		if ( !CKEDITOR.env.edge || error.number !== -2146828218 ) {
			throw( error );
		}
	}

	this.editor.fire( 'contentDomUnload' );

	delete this.editor.document;
	delete this.editor.window;
	delete this.editor;
},
```
and private helper:

````javascript
_: {
	detach: function() {
		// Update the editor cached data with current data.
		this.removeClass( 'cke_editable' );
		this.restoreAttrs();

		// Cleanup our custom classes.
		var classes;
		if ( ( classes = this.removeCustomData( 'classes' ) ) ) {
			while ( classes.length )
				this.removeClass( classes.pop() );
		}

		// Remove contents stylesheet from document if it's the last usage.
		if ( !this.is( 'textarea' ) ) {
			var doc = this.getDocument(),
				head = doc.getHead();
			if ( head.getCustomData( 'stylesheet' ) ) {
				var refs = doc.getCustomData( 'stylesheet_ref' );
				if ( !( --refs ) ) {
					doc.removeCustomData( 'stylesheet_ref' );
					var sheet = head.removeCustomData( 'stylesheet' );
					sheet.remove();
				} else {
					doc.setCustomData( 'stylesheet_ref', refs );
				}
			}
		}
	}
}
```

Also, renaming `_.detach` into something like `_.cleanUpCustomElementData` would improve readability.